### ISSUE TITLE
Reachable panic in `FileLike::metadata()`

### ISSUE THREAD

**Issue by Marsman1996** [asterinas/asterinas#1193]
### Describe the bug
There is a reachable panic in `FileLike::metadata()` at kernel/aster-nix/src/fs/file_handle.rs:58 when make a `fstat` syscall.

https://github.com/asterinas/asterinas/blob/0291b5dc6bb142b9c6165c1cb29b7658eefdaa63/kernel/aster-nix/src/fs/file_handle.rs#L57-L59

### To Reproduce
1. Compile a program which calls `fstat`:
```C
#include <errno.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <sys/mman.h>
#include <sys/socket.h>
#include <sys/stat.h>
#include <sys/syscall.h>
#include <unistd.h>

int main() {
  void *addr = mmap(NULL, 4096, PROT_READ, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
  int fd = socket(1, 0x5, 0);
  fstat(fd, addr);
  return EXIT_SUCCESS;
}

```
2. Run the compiled program in Asterinas.

### Expected behavior
Asterinas reports panic and is terminated.

### Environment
- Official docker asterinas/asterinas:0.6.2
- 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
- Asterinas version: main 0291b5dc

### Logs
```
~ # /root/fstat.c 
panicked at /root/asterinas/kernel/aster-nix/src/fs/file_handle.rs:58:9:
metadata unsupported
Printing stack trace:
   1: fn 0xffffffff887cced0 - pc 0xffffffff887ccee8 / registers:

     rax               0x12; rdx 0xffffffff889e5558; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f629e80;

   2: fn 0xffffffff887cccb0 - pc 0xffffffff887cce28 / registers:

     rax               0x12; rdx 0xffffffff889e5558; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f629e90;

   3: fn 0xffffffff88049000 - pc 0xffffffff8804900a / registers:

     rax               0x12; rdx 0xffffffff889e5558; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f62a010;

   4: fn 0xffffffff8895bd90 - pc 0xffffffff8895be12 / registers:

     rax               0x12; rdx 0xffffffff889e5558; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f62a020;

   5: fn 0xffffffff88134bd0 - pc 0xffffffff88134c00 / registers:

     rax               0x12; rdx 0xffffffff889e5558; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f62a0b0;

   6: fn 0xffffffff885c09f0 - pc 0xffffffff885c0d33 / registers:

     rax               0x12; rdx 0xffffffff889e5558; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f62a100;

   7: fn 0xffffffff885c0fb0 - pc 0xffffffff885c1ccf / registers:

     rax               0x12; rdx 0xffffffff889e5558; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f62a460;

   8: fn 0xffffffff88136b80 - pc 0xffffffff88164718 / registers:

     rax               0x12; rdx 0xffffffff889e5558; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f62ab40;

   9: fn 0xffffffff88121f60 - pc 0xffffffff88121fee / registers:

     rax               0x12; rdx 0xffffffff889e5558; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f6403d0;

  10: fn 0xffffffff88515020 - pc 0xffffffff88515b9f / registers:

     rax               0x12; rdx 0xffffffff889e5558; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f640570;

  11: fn 0xffffffff884522a0 - pc 0xffffffff884522ae / registers:

     rax               0x12; rdx 0xffffffff889e5558; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f640f90;

  12: fn 0xffffffff887a72f0 - pc 0xffffffff887a7306 / registers:

     rax               0x12; rdx 0xffffffff889e5558; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f640fb0;

  13: fn 0xffffffff88843bc0 - pc 0xffffffff88843c29 / registers:

     rax               0x12; rdx 0xffffffff889e5558; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f640fd0;

  14: fn                0x0 - pc                0x0 / registers:

     rax               0x12; rdx 0xffffffff889e5558; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f641000;

[OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panicking.rs:107
(  2) /root/asterinas/ostd/src/panicking.rs:59
(  3) 86161ztg45fpd9b1tjbn4gsl1:?
(  4) ??:?
(  5) /root/asterinas/kernel/aster-nix/src/fs/file_handle.rs:61
(  6) /root/asterinas/kernel/aster-nix/src/syscall/stat.rs:20
(  7) /root/asterinas/kernel/aster-nix/src/syscall/stat.rs:60
(  8) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:173
(  9) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:323
( 10) /root/asterinas/kernel/aster-nix/src/thread/task.rs:69
( 11) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:79
( 12) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2077
( 13) /root/asterinas/ostd/src/task/task/mod.rs:310
make: *** [Makefile:153: run] Error 1
```

**Comment by Marsman1996** [comment]
Is it a `todo` feature?

**Comment by grief8** [comment]
> Is it a `todo` feature?

Hi! It appears that the `FileLike::metadata()` method is currently unimplemented for the `Socket` type in Asterinas. This should be marked as a `todo` feature, and I'm wondering if there's anyone actively working on its implementation. Do you have any idea @StevenJiang1110 ?

**Comment by Marsman1996** [comment]
Fixed by #1520
